### PR TITLE
Add regression tests for coarse-graining configurations

### PR DESCRIPTION
**Pull request by spencerkclark**
This PR adds basic regression tests exercising model-level and pressure-level coarse-graining.  This will make it significantly easier for myself (and possibly others) to make performance-related refactors, e.g. as suggested in https://github.com/VulcanClimateModeling/fv3gfs-fortran/pull/116#issuecomment-756263520, and will also ensure that other code changes do not impact coarse-graining functionality.

In addition, there was a bug in the way I re-implemented the coarse-graining procedure for the `sfc_data` restart files which prevented things from working properly when the model was run with more than one rank per tile.  This PR fixes that issue, and also adds regression tests of the coarse-graining that explicitly check that the model produces identical results if it is run with a 1x1 domain layout or 1x2 domain layout.

Perhaps due to a memory issue, these tests cannot be run on CircleCI with more ranks.  On my VM I can get them to pass with a 2x2 layout for instance.

### CODE REVIEW

**Review by rheacangeo — APPROVED** [review]
Very cool addition to what is tested and the flexibility added. My first question was whether this significantly increases the turn around time for tests, but circleci indicates it took 35 minutes, which is still in the zone of what we see. I have a couple of small suggestions/comments, but this looks pretty good to go from my perspective! You may want to get Jeremy's feedback so he's away of the pytest changes. Nice work!

File: tests/pytest/run_files/submit_job.sh [hunk 1]
```diff
@@ -1,10 +1,9 @@
 #!/bin/bash
 
-
-HEAD=$1
+N_PROCESSES=$1
 RUNDIR=/rundir
 
 ulimit -s unlimited
-cp $HEAD/FV3/fv3.exe $RUNDIR/fv3.exe
+cp /FV3/fv3.exe $RUNDIR/fv3.exe
```
[the remark this answers is not in the document]
Comment (spencerkclark) [reply tests/pytest/run_files/submit_job.sh:7]: Indeed -- we were actually not using it in the first place.  E.g. there is no argument passed here in the code before this PR:
https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/622a368f2f29c1b0123d99b03306ae555d75842d/tests/pytest/test_regression.py#L121

File: tests/pytest/test_regression.py [hunk 2]
```diff
@@ -45,32 +44,48 @@ def image_runner(request):
...
+
+@pytest.mark.parametrize(
+    ("config_filename", "tag"), 
+    [
+        ("default.yml", "{version}"),
+        ("default.yml", "{version}-serialize"),
+        ("restart.yml", "{version}"),
+        ("restart.yml", "{version}-serialize"),
+        ("model-level-coarse-graining.yml", "{version}"),
+        ("pressure-level-coarse-graining.yml", "{version}")
```
[the remark this answers is not in the document]
Comment (spencerkclark) [reply tests/pytest/test_regression.py]: Oh great!  That should hopefully make these changes a little more palatable in terms of CI time (now in the net we'll only be adding 3 more model runs).

File: tests/pytest/run_files/job_jenkins_sarus [hunk 3]
```diff
@@ -5,7 +5,6 @@
 #SBATCH --constraint=mc
 #SBATCH --time=00:04:00
 #SBATCH --nodes=1
-#SBATCH --ntasks=6
```
[the remark this answers is not in the document]
Comment (spencerkclark) [reply tests/pytest/run_files/job_jenkins_sarus:8]: Gotcha -- that sounds good.  Thanks.

**Review by oliverwm1 — APPROVED** [review]
Great to have regression tests for this functionality, thanks for implementing! If CI cost is a concern, we could think about just replacing the current main regression tests with the versions that include coarse-graining. But the current time taken seems fine to me.

Agreed might be a good idea to check with Jeremy about the pytest changes.

**Comment by spencerkclark** [comment]
> If CI cost is a concern, we could think about just replacing the current main regression tests with the versions that include coarse-graining.

Indeed, it occurred to me we could do that, but at least in the short run I like seeing that the coarse-graining code is not changing the runtime behavior or results of the originally configured model at all.

File: tests/pytest/test_regression.py [hunk 4]
```diff
@@ -24,15 +24,14 @@
 config_filenames = os.listdir(CONFIG_DIR)
 
 
-@pytest.fixture(params=["{version}", "{version}-serialize"])
-def model_image_tag(request):
-    return request.param.format(version=request.config.getoption("--image_version"))
+@pytest.fixture
+def image_version(request):
+    return request.config.getoption("--image_version")
```
Comment (mcgibbon) [inline tests/pytest/test_regression.py:29]: Makes sense.

**Review by mcgibbon — APPROVED** [review]
LGTM!

**Pull request merged**